Phase5/Search/Tokenizer.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+
+namespace Phase5
+{
+    public class Tokenizer : ITokenizer
+    {
+        public HashSet<string> Tokenize(string doc)
+        {

```

Comment:
Sounds that you can implement this method simpler, It's little hard to understand what happened here !!